clients/java/signalr/src/main/java/com/microsoft/signalr/JsonHubProtocol.java

```diff
@@ -83,8 +84,21 @@ public TransferFormat getTransferFormat() {
                             break;
                         case "arguments":
                             if (target != null) {
-                                List<Class<?>> types = binder.getParameterTypes(target);
-                                arguments = bindArguments(reader, types);
+                                try {
+                                    List<Class<?>> types = binder.getParameterTypes(target);
```

Comment:
keep this line outside the try

Why? Because if `getParameterTypes` throws, the error recovery code will go haywire and start gobbling lots of JSON. Plus, we actually want an exception there to propagate out.

Reply:
We put it inside the try in .NET too. The only error that should be thrown from `getParameterTypes` is "no callback registered"

Reply:
Ok, if it's inside the try in .NET I guess we keep it in here too :)

Reply:
Can you add a test for the case that throws that exception? I'm most worried about the JSON reader recovering properly. Make sure to put the invalid information **before** the Invocation ID in the test JSON payload so we can verify that the reader "recovers" and still is able to read the invocation ID